packages/vite/src/node/config.ts

```diff
@@ -400,6 +385,46 @@ export async function resolveConfig(
   const BASE_URL = resolveBaseUrl(config.base, command === 'build', logger)
   const resolvedBuildOptions = resolveBuildOptions(config.build)
 
+  let cjsInclude = resolvedBuildOptions.commonjsOptions.include
+  if (cjsInclude !== undefined) {
+    const include = (cjsInclude = arraify(cjsInclude))
+
+    // HACK: This uses an object that looks/acts like a RegExp
+    // but is actually dynamic so that symlinked files can be
+    // analyzed by CommonJS as their symlinks are resolved.
+    let cachedLength = include.length
+    let cachedFilter = createFilter(include)
+    resolvedBuildOptions.commonjsOptions.include = {
+      // Only this method is needed to be `createFilter` compatible.
+      // It assumes the `include` array length only ever increases if mutated.
+      test(id: string) {
+        if (include.length !== cachedLength) {
+          cachedLength = include.length
+          cachedFilter = createFilter(include)
+        }
+        return cachedFilter(id)
+      },
+      // Ensure "instanceof RegExp" returns true.
+      __proto__: RegExp.prototype
+    } as any
+  }
+
+  // resolve alias with internal client alias
+  const resolvedAlias = normalizeAlias(
```

Comment:
I don't remember why `const resolvedAlias` and `const resolveOptions` declarations were moved down. We could probably move them back.. :)